### PR TITLE
Bolts

### PR DESCRIPTION
**Pull request by Kurfursten**
This was something I started before my surgery and I wanted to see if people liked the idea before I finished it. I haven't done any work on it since then and it probably doesn't compile yet! Let's see what emojis think.

* Make plasma bolts out of plasteel
* Can apply to an engine floor to turn it into bolted floor, you can lay down on bolted floor but you can't crawl onto it. You can remove the bolts but you need to be standing directly on top of the floor to do so. This new bolted floor is hostile architecture to stop crawling (see pic). It has some immediate applications in that you could put it under MULE flaps, but it's mostly because of a vault rework I had in mind (this allows you to lay down to dodge lasers but not crawl under them)
* You can use bolts on any unanchored structure to anchor it - people keep asking me for anchorable lockers and crates, this does that
* You can use wires on bolts to create the frankenbolt costume item

image

🆑 
* rscadd: Make bolts out of plasteel. You can apply these to reinforced floors to make them uncrawlable (ideal for flaps) or to any structure to make it immobile (e.g.: fuel tank, crate, locker, office chair)

### CODE REVIEW

**Comment by Eneocho** [comment]
You totally could crawl onto a tile with those bolts, but it would hurt a fair bit, like hitting your stomach with the corner of a table.

**Comment by Kurfursten** [comment]
These are made with the same technology that prevents people from climbing on tables.

**Comment by pedr69** [comment]
tackling onto this turf should break your arms

File: code/game/objects/structures.dm [hunk 1]
```diff
@@ -1,12 +1,30 @@
...
+	..()
+	if(hasbolts)
+		to_chat(user,"<span class='info'>This one is bolted into place.</span>")
 
 /obj/structure/blob_act(var/destroy = 0)
 	..()
 	if(destroy || (prob(50)))
 		qdel(src)
 
+/obj/structure/attackby(obj/item/I, mob/user)
```
Comment (pedr69) [inline code/game/objects/structures.dm:16]: nothing personal, `/obj/structure/bed/chair/vehicle/janicart`

Comment (Exxion) [reply code/game/objects/structures.dm]: I believe vehicles are anchored, but also this would be funny if it worked

Comment (Kurfursten) [reply code/game/objects/structures.dm]: You are correct, vehicles are already anchored so they can't be bolted. I do agree that bolting a vehicle would be very funny though.

I think your comment has an underlying insight that testing this code is going to be a bitch given the number of structures but I am going to do my best (when it's ready).

Comment (pedr69) [reply code/game/objects/structures.dm:16]: i skimmed through the list and the only thing that caught my eye is the giftwrap strange present

Comment (GreatBigFailure) [reply code/game/objects/structures.dm]: What happens if you bolt the singulo? Or Narsie? They're structures aren't they?

Comment (Exxion) [reply code/game/objects/structures.dm]: They're machinery, and also anchored.

Comment (Eneocho) [reply code/game/objects/structures.dm]: Singulos pulling stuff unanchors them by default, right? Can you unanchor a singulo via this process and bolt it?

Comment (Exxion) [reply code/game/objects/structures.dm]: They're machinery, not structures.

Comment (Eneocho) [reply code/game/objects/structures.dm]: oh, aight

**Comment by ShiftyRail** [comment]
do you intend to keep working on this, Kurf ?

**Comment by Kurfursten** [comment]
Yeah, I'll try to get around to this soon

**Comment by Kurfursten** [comment]
Tested...

Properly does work
Watertank
Potted plant
Office chair
Closet
Crate

Properly doesn't work (intended)
Window
Table
Janicart

**Comment by Eneocho** [comment]
Could you use the bolts to keep a telebeacon in place?

**Comment by Kurfursten** [comment]
Telebeacons are items, so no. This only works on structures. Possibly we could open it up to more stuff if this goes well?

Notable stuff this does NOT work on
* Items like telebeacon, igniter/signaller assembly, spare ID, plates
* Machinery like beepsky, gravitational singularity, nuclear bomb,

**Comment by hacker-on-steroids** [comment]
>you can tackle over them without breaking your arms

one job...

**Comment by Kurfursten** [comment]
Getting tackle-punishment to work to my satisfaction ended up being a huge pain and I wasn't sure how much people cared about it anyway
Maybe someone else will do it

**Pull request merged**